tests/unit/mathematics/linalg/LinalgBackendViennaCL_unittest.cc

```diff
@@ -88,13 +88,17 @@ TEST(LinalgBackendViennaCL, SGVector_clone)
 	sg_linalg->set_gpu_backend(new LinalgBackendViennaCL());
 
 	const index_t size = 10;
-	SGVector<int32_t> a(size), b(size), c(size);
+	SGVector<int32_t> a(size);
 	a.range_fill(0);
 
-	b = to_gpu(a);
+	auto b = to_gpu(a);
+	ASSERT_EQ(a.size(), b.size());
+
 	auto d = b.clone();
-	c = from_gpu(d);
+	ASSERT_EQ(b.size(), d.size());
 
+	auto c = from_gpu(d);
+	ASSERT_EQ(c.size(), d.size())
 
```

Comment:
Yes, as otherwise it might segfault when you try to read the matrix at these indices, (say if the create matrix was smaller for some reason)